lib/iris/fileformats/netcdf.py

```diff
@@ -1119,20 +1119,13 @@ def _cf_coord_identity(self, coord):
 
         units = str(coord.units)
 
-        # TODO: Use #61 to get the units.
-        if isinstance(coord.coord_system, iris.coord_systems.GeogCS):
-            if "latitude" in coord.standard_name:
+        if ((isinstance(coord.coord_system, iris.coord_systems.GeogCS) or
```

Comment:
perhaps an in-line comment here would help a little.  I would suggest

Set the 'units' of 'latitude' and 'longitude' coordinates specified in 'degrees' to 'degrees_north' and 'degrees_east' respectively, as defined in the CF conventions for netCDF files: sections 4.1 and 4.2.